extensions/ql-vscode/src/remote-queries/shared/variant-analysis.ts

```diff
@@ -78,6 +79,12 @@ export interface VariantAnalysisSkippedRepositoryGroup {
   }>
 }
 
+export interface VariantAnalysisScannedRepositoryResult {
+  repositoryId: number;
+  interpretedResults?: AnalysisAlert[];
+  rawResults?: AnalysisRawResults;
+}
+
```

Comment:
Thanks, that would make it more explicit. I don't think we even need the `resultsType` to make it explicit that only one of the two is defined.

Reply:
On second thought, now that I try implementing it, I think it might make it harder to pass down props. For example, it would make this much harder:

https://github.com/github/vscode-codeql/blob/f8cc3aec3201310fa0d0d7e4b68353ac21532d04/extensions/ql-vscode/src/view/variant-analysis/VariantAnalysisAnalyzedRepos.tsx#L41-L42

I would ideally like to keep those components independent from the exact structure of the `VariantAnalysisScannedRepositoryResult`, so I think I'll leave it like this for now.

Reply:
When I tried it out I changed those props to instead take a `VariantAnalysisScannedRepositoryResult`, but I agree perhaps that's making them a bit too tied to that type. So keeping things as they are is perfectly fine 👍